api/util.py

```diff
@@ -106,15 +117,18 @@ def get_meal_name(schedule: dict, meal_id: int) -> str:
 
         if meal_id == 1 and 'lunch' not in schedule:
             return 'brunch'
-    
+
     return MEAL_TO_PERIOD[meal_id][1]
-    
+
+
 def parse_date(date: str) -> time.struct_time:
     '''
     Parse the date string "Weekday, Month Day, Year"
     into time.struct_time object
     '''
-    return time.strptime(date, "%A, %B %d, %Y")
+    if len(date) > 0:
```

Comment:
I get that this prevents an exception when this function is given an empty string, but why is it receiving an empty string in the first place? I'm guessing that the _real_ bug is somewhere deeper in the parsing code, since this function should never be receiving an empty string in the first place. It's actually good that this function raises the exception since it's telling us in detail what's wrong, instead of just producing unexpected behavior. If it's receiving an empty string for valid reasons, then this fix is fine, but I want to know if the empty string isn't a bug itself before merging this.

Reply:
I don't think the empty string is a bug itself. It will be an empty string if the table itself is empty. Our webscraping part for the event find all the td regardless the string inside is empty or not that is why we will get empty string inputs. 
<img width="536" alt="image" src="https://user-images.githubusercontent.com/96160110/208541516-1b613ad3-c826-4e4c-bcd3-25753e516b3f.png">


There isn't an empty string here because the table is full of content: 
<img width="1746" alt="image" src="https://user-images.githubusercontent.com/96160110/208541203-bee6859a-cced-400e-8120-830ff65f77fc.png">

However, here it will have an empty string because the table is empty.

<img width="1750" alt="image" src="https://user-images.githubusercontent.com/96160110/208541352-fec5e771-5193-4d13-935f-61c77363070e.png">

Reply:
Ok that makes sense. However, I think the code would be more clear if you just explicitly said `if text_list[0]=='': return False` instead of putting that logic one function deeper.